bin/init.ml

```diff
@@ -167,86 +77,170 @@ let term =
       "A comma separated list of ppx preprocessors used by the component"
     in
     Arg.(value & opt (list atom_conv) [] & info [ "ppx" ] ~docv ~doc)
-  and+ public =
-    (* TODO(shonfeder): Move to subcommands {lib, exe} once implemented *)
-    let docv = "PUBLIC_NAME" in
-    let doc =
-      "If called with an argument, make the component public under the given \
-       PUBLIC_NAME. If supplied without an argument, use NAME."
-    in
-    Arg.(
-      value
-      & opt ~vopt:(Some Component.Options.Use_name) (some public_name_conv) None
-      & info [ "public" ] ~docv ~doc)
-  and+ inline_tests =
-    (* TODO(shonfeder): Move to subcommand [lib] once implemented *)
-    let docv = "USE_INLINE_TESTS" in
-    let doc =
-      "Whether to use inline tests. Only applicable for $(b,library) and \
-       $(b,project) components."
-    in
-    Arg.(value & flag & info [ "inline-tests" ] ~docv ~doc)
-  and+ template =
-    let docv = "PROJECT_KIND" in
-    let doc =
-      "The kind of project to initialize. Valid options are $(b,e[xecutable]) \
-       or $(b,l[ibrary]). Defaults to $(b,executable). Only applicable for \
-       $(b,project) components."
-    in
-    Arg.(
-      value
-      & opt (some (enum Component.Options.Project.Template.commands)) None
-      & info [ "kind" ] ~docv ~doc)
-  and+ pkg =
-    let docv = "PACKAGE_MANAGER" in
-    let doc =
-      "Which package manager to use. Valid options are $(b,o[pam]) or \
-       $(b,e[sy]). Defaults to $(b,opam). Only applicable for $(b,project) \
-       components."
-    in
-    Arg.(
-      value
-      & opt (some (enum Component.Options.Project.Pkg.commands)) None
-      & info [ "pkg" ] ~docv ~doc)
+  in
+  { Component.Options.Common.name; libraries; pps }
+
+let context : Init_context.t Term.t =
+  let+ common_term = Common.term_with_default_root_is_cwd
+  and+ path =
+    let docv = "PATH" in
+    Arg.(value & pos 1 (some string) None & info [] ~docv)
   in
   let config = Common.init common_term in
-  Dune_engine.Clflags.on_missing_dune_project_file := Dune_engine.Clflags.Ignore;
-  let open Component in
-  let context =
-    Scheduler.go ~common:common_term ~config (fun () ->
-        Memo.run (Init_context.make path))
+  Scheduler.go ~common:common_term ~config (fun () ->
+      Memo.run (Init_context.make path))
+
+let public : Component.Options.public_name option Term.t =
+  let docv = "PUBLIC_NAME" in
+  let doc =
+    "If called with an argument, make the component public under the given \
+     PUBLIC_NAME. If supplied without an argument, use NAME."
+  in
+  Arg.(
+    value
+    & opt ~vopt:(Some Component.Options.Use_name) (some public_name_conv) None
+    & info [ "public" ] ~docv ~doc)
+
+let inline_tests : bool Term.t =
+  (* TODO(shonfeder): Move to subcommand [lib] once implemented *)
```

Comment:
Is this addressed now?

Reply:
Indeed. Also a few other overlooked TODOs fixed in 260c663976ae191ee20328f7ddad571f1f2c7fe4